nav/support/sim.py

```
@@ -0,0 +1,466 @@
+
+    # ------------------------------------------------------------------
+    # 1. Radius distribution: power law in [R_min, R_max]
+    # ------------------------------------------------------------------
+    def power_law_radius(rng: np.random.RandomState,
+                         R_min: float,
+                         R_max: float,
+                         alpha: float,
+                         size: Optional[int] = None) -> NDArrayFloatType:
+        """Sample R from p(R) ∝ R^(-alpha) on [R_min, R_max], alpha > 1."""
+        if alpha <= 1:
+            raise ValueError("alpha must be > 1 for a proper power law.")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)